Parser/action_helpers.c

```diff
@@ -1691,3 +1691,18 @@ _PyPegen_concatenate_strings(Parser *p, asdl_expr_seq *strings,
     assert(current_pos == n_elements);
     return _PyAST_JoinedStr(values, lineno, col_offset, end_lineno, end_col_offset, p->arena);
 }
+
+stmt_ty
+_PyPegen_checked_future_import(Parser *p, identifier module, asdl_alias_seq * names, int level,
+                  			   int lineno, int col_offset, int end_lineno, int end_col_offset,
+                      		   PyArena *arena) {
+    if (PyUnicode_CompareWithASCIIString(module, "__future__") == 0) {
```

Comment:
To ignore relative imports, we can add this check also used in #118216:

```suggestion
    if (level == 0 && PyUnicode_CompareWithASCIIString(module, "__future__") == 0) {
```

Reply:
@JelleZijlstra I think this addresses the relative import concerns. I will add a test soon